op-node/rollup/derive/batches_test.go

```diff
@@ -651,6 +652,30 @@ func TestValidBatch(t *testing.T) {
 		},
 	}
 
+	// Add test cases for all forks from Jovian to assert that upgrade block must not contain user
+	// txs. If a future fork should allow user txs in its upgrade block, it must be removed from
+	// this list explicitly.
+	for _, fork := range rollup.ForksFrom(rollup.Jovian) {
+		singularBatchTestCases = append(singularBatchTestCases, ValidBatchTestCase{
+			Name:       fmt.Sprintf("user txs in %s upgrade block", fork),
+			L1Blocks:   []eth.L1BlockRef{l1A, l1B, l1C},
+			L2SafeHead: l2A0,
+			Batch: BatchWithL1InclusionBlock{
+				L1InclusionBlock: l1B,
+				Batch: &SingularBatch{
+					ParentHash:   l2A1.ParentHash,
+					EpochNum:     rollup.Epoch(l2A1.L1Origin.Number),
+					EpochHash:    l2A1.L1Origin.Hash,
+					Timestamp:    l2A1.Time,
+					Transactions: []hexutil.Bytes{[]byte("invalid tx in upgrade block")},
```

Comment:
```suggestion
					Transactions: []hexutil.Bytes{[]byte("tx in upgrade block")},
```
Nit: I slightly prefer this to emphasise the the batch is considered invalid even if the tx is not invalid per se. Any tx whatsoever would make the batch invalid.

Reply:
it's just some random text... but I'll reword